networktable.c

```diff
@@ -76,16 +167,35 @@ void parse_arp_cache(void)
 	// Read ARP cache line by line
 	while(getline(&linebuffer, &linebuffersize, arpfp) != -1)
 	{
-		int num = sscanf(linebuffer, "%99s 0x%x 0x%x %99s %99s %99s\n",
-		                 ip, &type, &flags, hwaddr, mask, iface);
-
-		// Skip header and empty lines
-		if (num < 4)
-			continue;
+		int num = sscanf(linebuffer, "%99s dev %99s lladdr %99s %99s %99s",
+		                 ip, iface, hwaddr, status1, status2);
 
-		// Skip incomplete entires, i.e., entries without C (complete) flag
-		if(!(flags & 0x02))
-			continue;
+		// Check if we want to process the line we just read
+		switch(num)
+		{
+			// Example: 2003:17:7:dc:e1b1:1e01:1a5a:ff dev eth0 lladdr a0:20:29:24:98:99 REACHABLE
+			case 4:
+				if(strcasecmp(status1, "reachable") != 0 &&
+				   strcasecmp(status1, "stale")     != 0)
+				{
+					continue;
+				}
+				break;
+
+			// Example: dead:beef:dead:beef:dead:beef dev wlp3s0 lladdr de:ad:be:ef:de:ad router REACHABLE
+			case 5:
+				if(strcasecmp(status1, "router")    != 0 &&
+				   strcasecmp(status2, "reachable") != 0 &&
+				   strcasecmp(status2, "stale")     != 0)
+				{
+					continue;
+				}
+				break;
+
+			// Skip lines without hardware information
+			default:
+				continue;
+		}
```

Comment:
Would the logic be simpler if we only checked for the lines with the status that we want?

There is also a `nud` option which can filter by status (see the man page for `ip-neighbour`:
```
nud STATE
                     only list neighbour entries in this state.  NUD_STATE takes values listed below or the special value all which means all states. This option may occur more than once.  If this option
                     is absent, ip lists all entries except for none and noarp.
```